File: py/barang.py
from produk import Produk

class Barang(Produk):
    def __init__(self):
        super().__init__()
        self.merek = ""
        self.stok = 0
        self.tahun = 0

    # Getter
    def get_merek(self):
        return self.merek

    def get_stok(self):
        return self.stok

    def get_tahun(self):
        return self.tahun

    # Setter
    def set_merek(self, merek):
        self.merek = merek

    def set_stok(self, stok):
        self.stok = stok

    def set_tahun(self, tahun):
        self.tahun = tahun

    # Destructor
    def __del__(self):
        pass

File: php/barang.php
<?php

require_once 'produk.php';

class Barang extends Produk 
{
    public $merek;
    public $tahun;
    public $stok;

    public function __construct($id = "", $nama = "", $harga = 0, $stok = 0, $merek = "", $tahun = 0, $foto_produk = "") 
    {
        parent::__construct($id, $nama, $harga, $foto_produk);
        $this->stok = $stok;
        $this->merek = $merek;
        $this->tahun = $tahun;
    }

    // Getter
    public function getStok() { return $this->stok; }
    public function getMerek() { return $this->merek; }
    public function getTahun() { return $this->tahun; }
    // Setter
    public function setStok($stok) { $this->stok = $stok; }
    public function setMerek($merek) { $this->merek = $merek; }
    public function setTahun($tahun) { $this->tahun = $tahun; }
}